bookkeeper-server/src/main/java/org/apache/bookkeeper/replication/Auditor.java

```diff
@@ -998,6 +1006,13 @@ public void run() {
         }), initialDelay, interval, TimeUnit.SECONDS);
     }
 
+    private class UnderReplicatedLedgersChangedCb implements GenericCallback<Void> {
+        @Override
+        public void operationComplete(int rc, Void result) {
+            numUnderReplicatedLedger.registerSuccessfulValue(-1);
```

Comment:
Please double check me, my understanding is:
numUnderReplicatedLedger is an OpStatsLogger. This works for registerSuccessfulValue only with Prometheus, afaict Codahale simply treats it as a duration in millis (similar to the registerSuccessfulEvent but with time init set in millis) and adds into a histogram.
But that's a side problem.

If your intent is to "know the progress of replication from the stat" then you are getting into a situation when the metric can increase and decrease during the reporting interval and you won't see the progress.
I'd suggest to have a gauge with number of currently under replicated ledgers (numUnderreplicatedLedgersElapsedRecoveryGracePeriod is doing this right now) and a counter incremented for successfully replicated ledgers. Note that ledger can get out of under replicated state by either being re-replicated or if a missing bookie returns back into the cluster.

Reply:
Zookeeper Watcher added, the metric will be updated  When under replicated ledger path  deleted.
I

Reply:
I will try to add a new gauge with number of currently under replicated ledgers.

Reply:
I added three new metrics.  `UNDER_REPLICATED_LEDGERS_COUNT UNDER_REPLICATED_LEDGERS_SIZE REPLICATED_LEDGERS_COUNT` 
previous metrics are named `NUM_UNDER_REPLICATED_LEDGERS UNDER_REPLICATED_LEDGERS_TOTAL_SIZE`
I prefer to delete the previous OpStatsLogger metrics. so only `REPLICATED_LEDGERS_COUNT` will be added.

Reply:
@dlg99